org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/builder/MavenBuilderImpl.java

```diff
@@ -219,7 +223,12 @@ private boolean hasRelevantDelta(IMavenProjectFacade projectFacade, IResourceDel
         IPath fullPath = delta.getFullPath();
         if(buildOutputLocation.isPrefixOf(fullPath)) {
           //anything in the build output is not interesting for a change as it is produced by the build
-          //lets see if there are more interesting parts...
+          // ... unless a classpath resource that existed before has been deleted, possibly by another builder
+          if(!resource.exists() && ((outputLocation != null && outputLocation.isPrefixOf(fullPath))
```

Comment:
Please split this into smaller parts or maybe a method as such nested conditions are hard to read and understand, for example the frist and the second only differ in the location,